test/server/server_test.cc

```diff
@@ -92,27 +91,30 @@ class RunHelperTest : public testing::Test {
 };
 
 TEST_F(RunHelperTest, Normal) {
-  EXPECT_CALL(start_workers_, ready());
+  EXPECT_CALL(init_watcher_, ready());
   cm_init_callback_();
 }
 
 TEST_F(RunHelperTest, ShutdownBeforeCmInitialize) {
-  EXPECT_CALL(start_workers_, ready()).Times(0);
+  EXPECT_CALL(init_watcher_, ready()).Times(0);
   sigterm_->callback_();
   EXPECT_CALL(server_, isShutdown()).WillOnce(Return(shutdown_));
   cm_init_callback_();
 }
 
-TEST_F(RunHelperTest, ShutdownBeforeInitManagerInit) {
-  EXPECT_CALL(start_workers_, ready()).Times(0);
-  Init::MockTarget target;
-  init_manager_.registerTarget(target, "");
-  EXPECT_CALL(target, initialize(_));
-  cm_init_callback_();
-  sigterm_->callback_();
-  EXPECT_CALL(server_, isShutdown()).WillOnce(Return(shutdown_));
-  target.callback_();
-}
+// TODO(mergeconflict): this test isn't valid anymore, since the "if (!shutdown) startWorkers()"
```

Comment:
Let's either fix this now or open an issue and assign to yourself for follow up, prefer not to leave commented out code.

Reply:
Yeah, definitely agree. I just need to figure out how I want to fix this...